applications/nrf5340_audio/src/audio/streamctrl.c

```diff
@@ -470,13 +491,62 @@ static void le_audio_msg_sub_thread(void)
 
 		default:
 			LOG_WRN("Unexpected/unhandled le_audio event: %d", event);
+
 			break;
 		}
 
 		STACK_USAGE_PRINT("le_audio_msg_thread", &le_audio_msg_sub_thread_data);
 	}
 }
 
+/* Handle content control events */
+static void content_control_msg_sub_thread(void)
+{
+	int ret;
+
+	const struct zbus_channel *chan;
+
+	while (1) {
+		ret = zbus_sub_wait(&content_control_evt_sub, &chan, K_FOREVER);
+		ERR_CHK(ret);
+
+		struct content_control_msg msg;
+
+		ret = zbus_chan_read(chan, &msg, ZBUS_READ_TIMEOUT_MS);
+		ERR_CHK(ret);
+
+		uint8_t event = msg.event;
+
+		LOG_DBG("Received event = %d, current state = %d", event, strm_state);
+
+		switch (event) {
+		case MEDIA_PLAY:
+			ret = le_audio_play();
+			if (ret) {
+				LOG_ERR("Failed to play");
+			}
+
+			break;
+
+		case MEDIA_PAUSE:
+			ret = le_audio_pause();
+			if (ret) {
+				LOG_ERR("Failed to pause");
+			}
+
+			break;
+
+		default:
+			LOG_WRN("Unexpected/unhandled event: %d", event);
+
+			break;
+		}
+
+		STACK_USAGE_PRINT("content_control_msg_thread",
+				  &content_control_msg_sub_thread_data);
```

Comment:
It is purely for printing the stack usage, and only valid if CONFIG_THREAD_ANALYSER is enabled.

Reply:
it's only printed every CONFIG_PRINT_STACK_USAGE_MS as seen here: https://github.com/nrfconnect/sdk-nrf/blob/8afd7658bf244c3bf3a5eb6e3c0eff8b8df0f40a/applications/nrf5340_audio/src/utils/macros/macros_common.h#L43

Reply:
Ah, so a macro called `STACK_USAGE_PRINT` only conditionally prints something based on an invisible value diff (`k_uptime_get() - thread_ts`) that is also undocumented? :D 

Is there any reason why you've implemented this behavior yourself, rather than relying on the existing `thread_analyzer_print` and `thread_analyzer_auto`?

Reply:
It was done a while ago, and we haven't prioritized checking if it's still needed. It might not be